internal/transactionsubmission/engine/signing/distribution_account_env_signature_client.go

```diff
@@ -0,0 +1,129 @@
+package signing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/strkey"
+	"github.com/stellar/go/txnbuild"
+)
+
+type DistributionAccountEnvOptions struct {
+	DistributionPrivateKey string
+	NetworkPassphrase      string
+}
+
+func (opts *DistributionAccountEnvOptions) String() string {
+	return fmt.Sprintf("%T{NetworkPassphrase: %s}", opts, opts.NetworkPassphrase)
+}
+
+func (opts *DistributionAccountEnvOptions) Validate() error {
+	if opts.NetworkPassphrase == "" {
+		return fmt.Errorf("network passphrase cannot be empty")
+	}
+
+	if !strkey.IsValidEd25519SecretSeed(opts.DistributionPrivateKey) {
+		return fmt.Errorf("distribution private key is not a valid Ed25519 secret")
+	}
+
+	return nil
+}
+
+type DistributionAccountEnvSignatureClient struct {
+	networkPassphrase   string
+	distributionAccount string
+	distributionKP      *keypair.Full
+}
+
+func (c *DistributionAccountEnvSignatureClient) String() string {
+	return fmt.Sprintf("DistributionAccountEnvSignatureClient{distributionAccount: %s, networkPassphrase: %s}", c.distributionAccount, c.networkPassphrase)
+}
+
+// NewDistributionAccountEnvSignatureClient returns a new DistributionAccountEnvSignatureClient instance
+func NewDistributionAccountEnvSignatureClient(opts DistributionAccountEnvOptions) (*DistributionAccountEnvSignatureClient, error) {
+	if err := opts.Validate(); err != nil {
+		return nil, fmt.Errorf("validating options: %w", err)
+	}
+
+	distributionKP, err := keypair.ParseFull(opts.DistributionPrivateKey)
+	if err != nil {
+		return nil, fmt.Errorf("parsing distribution seed: %w", err)
+	}
+
+	return &DistributionAccountEnvSignatureClient{
+		networkPassphrase:   opts.NetworkPassphrase,
+		distributionAccount: distributionKP.Address(),
+		distributionKP:      distributionKP,
+	}, nil
+}
+
+var _ SignatureClient = (*DistributionAccountEnvSignatureClient)(nil)
+
+func (c *DistributionAccountEnvSignatureClient) SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error) {
+	if stellarTx == nil {
+		return nil, fmt.Errorf("stellarTx cannot be nil in %s", c.Type())
+	}
+
+	// Ensure that the distribution account is the only account signing the transaction
+	for _, stellarAccount := range stellarAccounts {
+		if stellarAccount != c.distributionAccount {
+			return nil, fmt.Errorf("stellar account %s is not allowed to sign in %s", stellarAccount, c.Type())
+		}
+	}
```

Comment:
What about moving this to a private function?

Reply:
Good idea, done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/7a02ff18489ce2aa017b64f46bee7703f7f3e27d.